src/main/java/com/bootcamp/demo/controller/AccountController.java

```diff
@@ -26,6 +30,37 @@ public AccountController(Service<Account> service) {
         }
     }
 
+    @PutMapping("/editAccount")
+    ResponseEntity<Account> update(@RequestBody AccountDetails accountDetails) throws ControllerException {
+        try {
+            Account searchedAccount = service.findOne(accountDetails.getUsername());
+
+            if (searchedAccount != null) {
+                Account account = new Account(
+                        accountDetails.getName(),
+                        accountDetails.getEmail(),
+                        accountDetails.getUsername(),
+                        searchedAccount.getPassword(),
+                        accountDetails.getPhoneNumber(),
+                        accountDetails.getAddress(),
+                        accountDetails.getDateOfBirth()
+                );
+
+                if (!Objects.equals(searchedAccount.getEmail(), accountDetails.getEmail())) {
+                    // TODO: Find a solution to return the error message: "Email cannot be changed!"
+                    return ResponseEntity.status(400).body(null);
+                }
+
+                return  ResponseEntity.status(200).body(this.service.update(account));
+            }
+        } catch (ServiceException | InterruptedException | ExecutionException exception) {
+            throw new ControllerException(exception.getMessage());
```

Comment:
No one is going to catch this exception. You should return a ResponseEntity that encapsulates the exception message and an appropriate http status.